src/Tools/dotnet-monitor/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.Egress;
+using Microsoft.Diagnostics.Monitoring.Egress.AzureStorage;
+using Microsoft.Diagnostics.Monitoring.Egress.FileSystem;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using System;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    internal static class ServiceCollectionExtensions
+    {
+        public static IServiceCollection ConfigureEgress(this IServiceCollection services, IConfiguration configuration)
+        {
+            // Configure AzureStorageOptions to bind to the AzureStorage configuration key.
+            services.Configure<AzureStorageOptions>(configuration.GetSection(AzureStorageOptions.ConfigurationKey));
+
+            IConfigurationSection egressSection = configuration.GetSection(EgressOptions.ConfigurationKey);
+
+            // Register change token for EgressOptions binding
+            services.AddSingleton((IOptionsChangeTokenSource<EgressOptions>)new ConfigurationChangeTokenSource<EgressOptions>(egressSection));
```

Comment:
Did you have a chance to validate that a changed mounted settings file in k8 will probably trigger the change token?

Reply:
Not yet; my understanding is that the library that provides key-per-file configuration only started supporting change notification in the 5.0 timeframe. Not sure we want to depend on the rc2 version of the library yet or just wait for RTM.